vision/pipeline.py

```diff
@@ -112,25 +113,42 @@ def run(self, prev_state):
             if flags.obstacle_finder:
                 try:
                     self.obstacle_tracker.update(bboxes)
-                    bboxes = self.obstacle_tracker.getPersistentObstacles()
+                    bboxes = self.obstacle_tracker.get_persistent_obstacles()
                 except:
                     flags.obstacle_tracker = False
 
         elif state == "text_detection":  # approaching mast
             flags = TextDetectionFlags()
 
             try:
-                bboxes = self.text_detector.detect_russian_word(color_image, depth_image)
+                bboxes = self.text_detector.detect_russian_word(
+                    color_image, depth_image
+                )
             except:
                 flags.detect_russian_word = False
 
         elif state == "module_detection":  # locating module
             flags = ModuleDetectionFlags()
             try:
-                self.module_location.setImg(color_image, depth_image)
+                self.module_location.set_img(color_image, depth_image)
             except:
                 flags.set_img = False
 
+            if flags.set_img:
+                try:
+                    text_bboxes = self.text_detector.detect_russian_word(
+                        color_image, depth_image
+                    )
```

Comment:
Since you are now getting bounding boxes for text in this state you should append these to bboxes as well. That way, there will be both text and module bounding boxes for flight to work with.